### ISSUE TITLE
Jinja UndefinedError with header tags

### ISSUE THREAD

**Issue by NiklasRosenstein** [squidfunk/mkdocs-material#142]
I get this error when I place a `<` or `>` in a Markdown header, like

```
# <some.hpp>
```

Or even when I juse HTML tags like

```
<h2 id="nr.pvrq2.RenderJob">RenderJob</h2>
```

--- 

Error traceback:

```
  File "c:\users\niklas\repos\pydoc-markdown\.env\lib\site-packages\mkdocs\__main__.py", line 127, in serve_command
    livereload=livereload
  File "c:\users\niklas\repos\pydoc-markdown\.env\lib\site-packages\mkdocs\commands\serve.py", line 82, in serve
    config = builder()
  File "c:\users\niklas\repos\pydoc-markdown\.env\lib\site-packages\mkdocs\commands\serve.py", line 78, in builder
    build(config, live_server=live_server, dirty=dirty)
  File "c:\users\niklas\repos\pydoc-markdown\.env\lib\site-packages\mkdocs\commands\build.py", line 379, in build
    build_pages(config, dirty=dirty)
  File "c:\users\niklas\repos\pydoc-markdown\.env\lib\site-packages\mkdocs\commands\build.py", line 332, in build_pages
    dump_json)
  File "c:\users\niklas\repos\pydoc-markdown\.env\lib\site-packages\mkdocs\commands\build.py", line 212, in _build_page
    output_content = template.render(context)
  File "c:\users\niklas\repos\pydoc-markdown\.env\lib\site-packages\jinja2\environment.py", line 989, in render
    return self.environment.handle_exception(exc_info, True)
  File "c:\users\niklas\repos\pydoc-markdown\.env\lib\site-packages\jinja2\environment.py", line 754, in handle_exception
    reraise(exc_type, exc_value, tb)
  File "c:\users\niklas\repos\pydoc-markdown\.env\lib\site-packages\jinja2\_compat.py", line 37, in reraise
    raise value.with_traceback(tb)
  File "c:\users\niklas\repos\pydoc-markdown\.env\lib\site-packages\material\main.html", line 1, in top-level template code
    {% extends "base.html" %}
  File "c:\users\niklas\repos\pydoc-markdown\.env\lib\site-packages\material\base.html", line 87, in top-level template code
    {% block site_nav %}
  File "c:\users\niklas\repos\pydoc-markdown\.env\lib\site-packages\material\base.html", line 92, in block "site_nav"
    {% include "partials/nav.html" %}
  File "c:\users\niklas\repos\pydoc-markdown\.env\lib\site-packages\material\partials\nav.html", line 18, in top-level template code
    {% include "partials/nav-item.html" %}
  File "c:\users\niklas\repos\pydoc-markdown\.env\lib\site-packages\material\partials\nav-item.html", line 19, in top-level template code
    {% include "partials/nav-item.html"  %}
  File "c:\users\niklas\repos\pydoc-markdown\.env\lib\site-packages\material\partials\nav-item.html", line 29, in top-level template code
    {% set toc_ = (toc_ | first).children %}
  File "c:\users\niklas\repos\pydoc-markdown\.env\lib\site-packages\jinja2\environment.py", line 408, in getattr
    return getattr(obj, attribute)
jinja2.exceptions.UndefinedError: No first item, sequence was empty.
```

**Comment by squidfunk** [comment]
Please provide the complete source of the page which is causing the issue (you can replace with lorem ipsum, though) and some information on versions. See the issue template for this reason.

**Comment by NiklasRosenstein** [comment]
__index.md__

```md
<h1 id="nr.pvrq2">nr.pvrq2</h1>

FooBar Baz
```

__mkdocs.yml__

```yml
site_name: My Docs
theme: material
```

__Error__

```
[...]
  File "c:\python35-32\lib\site-packages\mkdocs\commands\build.py", line 212, in _build_page
    output_content = template.render(context)
  File "c:\python35-32\lib\site-packages\jinja2\environment.py", line 989, in render
    return self.environment.handle_exception(exc_info, True)
  File "c:\python35-32\lib\site-packages\jinja2\environment.py", line 754, in handle_exception
    reraise(exc_type, exc_value, tb)
  File "c:\python35-32\lib\site-packages\jinja2\_compat.py", line 37, in reraise
    raise value.with_traceback(tb)
  File "c:\python35-32\lib\site-packages\material\main.html", line 1, in top-level template code
    {% extends "base.html" %}
  File "c:\python35-32\lib\site-packages\material\base.html", line 87, in top-level template code
    {% block site_nav %}
  File "c:\python35-32\lib\site-packages\material\base.html", line 92, in block "site_nav"
    {% include "partials/nav.html" %}
  File "c:\python35-32\lib\site-packages\material\partials\nav.html", line 18, in top-level template code
    {% include "partials/nav-item.html" %}
  File "c:\python35-32\lib\site-packages\material\partials\nav-item.html", line 29, in top-level template code
    {% set toc_ = (toc_ | first).children %}
  File "c:\python35-32\lib\site-packages\jinja2\environment.py", line 408, in getattr
    return getattr(obj, attribute)
jinja2.exceptions.UndefinedError: No first item, sequence was empty.
```

**Comment by NiklasRosenstein** [comment]
I also just tried it with `1.0.3`

**Comment by NiklasRosenstein** [comment]
Actually I can not reproduce the `# <some.hpp>` example anymore, so ignore that for now. But the HTML header tags is reproducible.

**Comment by squidfunk** [comment]
Well, it has to do with the table of contents extension. The TOC cannot be generated when you use HTML tags, only when using the Markdown. The error shouldn't be there and there should be no TOC. Will fix it as soon as I find the time.

**Comment by NiklasRosenstein** [comment]
> The TOC cannot be generated when you use HTML tags, only when using the Markdown. The error shouldn't be there and there should be no TOC. Will fix it as soon as I find the time.

By the way, do you know why that is? Is it a limitation of MkDocs or a design decision? I need to generate HTML tags to assign them correct IDs, and eventually additional styling (eg. parts of the header in a smaller font and/or in monospace).

**Comment by squidfunk** [comment]
This is not a limitation of MkDocs, but of the Python Markdown Extensions MkDocs uses. Monospaced and smaller text is possible and supported by Material:

```
# Header with `code` and <small>small text</small>
```

You could add your anchor as HTML in the header, though:

```
# <a id="nr.pvrq2"></a> nr.pvrq2
```

Untested, may break other stuff.

**Comment by squidfunk** [comment]
I now know why it breaks - I had to apply a hack to check whether the content contains a `h1` headline to see if the page title should be set as a fallback (if it doesn't) and this matches your code exactly. You can fix it temporarily by adding another tag in front of the `h1`:

``` html
<h1 class="" id="nr.pvrq2">nr.pvrq2</h1>
```

I will provide a fix.

**Comment by squidfunk** [comment]
Fix https://github.com/squidfunk/mkdocs-material/pull/144 is in `master` now. Please checkout the `master` and confirm. Instructions on how to use Material from `master` are here (2 easy steps):

1. http://squidfunk.github.io/mkdocs-material/getting-started/#cloning-from-github
2. http://squidfunk.github.io/mkdocs-material/getting-started/#usage

**Comment by squidfunk** [comment]
Released in 1.0.4

**Comment by NiklasRosenstein** [comment]
Thanks, I can confirm it's working now! :)
